website/docs/sources/discord/index.md

```diff
@@ -202,6 +202,11 @@ guild_member_info = requests.get(
   },
 ).json()
 
+# Ensure we are not being ratelimited
+if guild_member_info.get("message") == "You are being rate limited.":
```

Comment:
This is quite weak as a check. It would be better to:

```python3
guild_member_request = requests.get(
  GUILD_API_URL.format(guild_id=guild_id),
  headers={
    "Authorization": f"Bearer {access_token}",
  },
)
guild_member_info = guild_member_request.json()

# Ensure we are not being ratelimited
if guild_member_request.status_code == 429:
	pass # do things
```

And the same below.

Cf. https://discord.com/developers/docs/topics/rate-limits

Reply:
I am not sure if guild_member_request has this info. When I sent guild_member_info as an event it only had the following properties:
```
"guild_member_info": {
        "global": false,
        "message": "You are being rate limited.",
        "retry_after": 35.683
    },
```

Reply:
So it goes like this:
```python
guild_member_request = requests.get(...)
# This is a requests.Response object, which has, among a lot of other things, a status_code attribute and a json() method.
guild_member_info = guild_member_request.json()
# This takes the requests.Response.body and passes it to json.loads and gives it back.
```
So from `guild_member_request`, you can get the HTTP status code to check you're not being rate limited, and then from `guild_member_info` you can get either the "roles" key (and others) if you're not being rate limited, or the `retry_after` key if you are.